server/athenian/api/controllers/filter_controller.py

```diff
@@ -42,16 +43,24 @@
 async def filter_contributors(request: AthenianWebRequest, body: dict) -> web.Response:
     """Find developers that made an action within the given timeframe."""
     try:
-        filt = GenericFilterRequest.from_dict(body)  # type: GenericFilterRequest
+        filt = FilterContributorsRequest.from_dict(body)  # type: FilterContributorsRequest
     except ValueError as e:
         # for example, passing a date with day=32
         return ResponseError(InvalidRequestError("?", detail=str(e))).response
     repos = await _common_filter_preprocess(filt, request)
     users = await mine_contributors(
-        repos, filt.date_from, filt.date_to, request.mdb, request.cache)
+        repos, filt.date_from, filt.date_to, request.mdb, request.cache,
+        with_roles=filt.as_)
     model = [
-        DeveloperSummary(login=f"{PREFIXES['github']}{u['login']}", avatar=u["avatar_url"],
-                         name=u["name"], updates=DeveloperUpdates(**u["stats"]))
+        DeveloperSummary(
+            login=f"{PREFIXES['github']}{u['login']}", avatar=u["avatar_url"],
+            name=u["name"], updates=DeveloperUpdates(**{
+                k: v for k, v in u["stats"].items()
+                # It worth make `DeveloperUpdates` support all the stats we can get
+                # instead of doing this filtering. See also `mine_contributors`.
```

Comment:
```suggestion
                # TODO(se7entyse7en): make `DeveloperUpdates` support all the stats we can get instead of doing this filtering. See also `mine_contributors`.  # noqa
```